modules/photons_control/clean.py

```diff
@@ -0,0 +1,189 @@
+from delfick_project.norms.spec_base import NotSpecified
+from photons_control.script import FromGenerator
+
+from photons_app.tasks import task_register as task
+
+from photons_messages import LightMessages
+
+from delfick_project.norms import sb, Meta
+
+
+def SetCleanConfig(indication, duration, reference=None, **kwargs):
+    """
+    Returns a valid message that will set the default clean cycle configuration
+    for the device.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(SetCleanConfig(indication=True), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(ref, sender, **kwargs):
+        r = ref if reference is None else reference
+
+        plans = sender.make_plans("capability")
+        async for serial, _, info in sender.gatherer.gather(plans, r, **kwargs):
+            if info["cap"].has_hev:
+                yield LightMessages.SetHevCycleConfiguration(
+                    indication=bool(indication),
+                    duration_s=duration,
+                    target=serial,
+                    ack_required=True,
+                    res_required=False,
+                )
+
+    return FromGenerator(gen)
+
+
+@task
+class set_clean_default_config(task.Task):
+    """
+    Set the default clean cycle configuration for a device
+
+    ``target:clean_config match:cap=hev -- '{"indication": true, "duration": 7200}'``
+
+    It takes an "indication" field that is a boolen which indicates whether to
+    run a short flashing indication at the end of the HEV cycle. It also takes
+    a ``duration`` field that is the seconds of the duration.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+        options = sb.set_options(
+            indication=sb.required(sb.boolean()),
+            duration=sb.required(sb.integer_spec()),
+        ).normalise(Meta.empty(), options)
+
+        await self.target.send(SetCleanConfig(**options), self.reference, **kwargs)
+
+
+@task
+class get_clean_default_config(task.Task):
+    """
+    Get the default HEV cleaning cycle configuration for the target bulb
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        async def gen(reference, sender, **kwargs):
```

Comment:
A "CleanPlan" would be cool for this https://github.com/delfick/photons/blob/daa74a3aaac94b25816bfd36ce8036cb89916995/modules/photons_control/multizone.py#L34 that returns current configuration and status

Reply:
I called it a `HevCyclePlan` but `CleanPlan` is better. :) I'll change it.

Reply:
I created a plan for my interactor commands, which will need some minor refactoring to work for this too. On the todo list.

Reply:
oooh shiny.